indexing-service/src/main/java/io/druid/indexing/common/task/IndexTask.java

```diff
@@ -416,6 +417,15 @@ public TaskStatus run(final TaskToolbox toolbox) throws Exception
         ((IngestSegmentFirehoseFactory) firehoseFactory).setTaskToolbox(toolbox);
       }
 
+      if (firehoseFactory instanceof CombiningFirehoseFactory) {
```

Comment:
What if a Combining firehose has another Combining firehose within it? This should be recursive.